proxy/envoy/discovery.go

```diff
@@ -194,13 +195,15 @@ type DiscoveryServiceOptions struct {
 	Port            int
 	EnableProfiling bool
 	EnableCaching   bool
+	EnableLocal     bool
 }
 
 // NewDiscoveryService creates an Envoy discovery service on a given port
 func NewDiscoveryService(ctl model.Controller, configCache model.ConfigStoreCache,
 	environment proxy.Environment, o DiscoveryServiceOptions) (*DiscoveryService, error) {
 	out := &DiscoveryService{
 		Environment: environment,
+		useVirtual:  !o.EnableLocal,
```

Comment:
I would rename the useVirtual flag to enableLocal and use the same naming everywhere.

If I understand correctly, this distinguishes the case where iptable redirection is used, so I am not sure if the word "virtual" is the most appropriate.